worlds/rabi_ribi/__init__.py

```diff
@@ -73,20 +76,25 @@ class RabiRibiWorld(World):
     item_name_to_id = item_table
     location_name_to_id: Dict[str, int] = all_locations
 
+    start_location: str
     picked_templates: List[str]
+    total_locations: int
     map_transition_shuffle_order: List[int]
     map_transition_shuffle_spoiler: List[str]
-
-    def __init__(self, multiworld, player):
-        super().__init__(multiworld, player)
-        self.total_locations = 0
+    existing_randomizer_args: Any
+    randomizer_data: RandomizerData
 
     def generate_early(self) -> None:
         """Set world specific generation properties"""
+        self.existing_randomizer_args = self._convert_options_to_existing_randomizer_args()
+        self.randomizer_data = RandomizerData(self.existing_randomizer_args)
 
         # Will be configurable later, but for now always force eggs to be local.
         self.options.local_items.value.add(ItemName.easter_egg)
 
+        # Force consumable items to be local, as the player may need to pick them up multiple times.
+        self.options.local_items.value.update(item_groups["Consumables"])
```

Comment:
> I assume the player can pick up consumables multiple times? How does that work?

For now, I moved Consumable items (Rumi Donut, Rumi Cake, Cocoa Bomb, Gold Carrot) to only be shuffled to the player locally. In the future, we might consider editing the map when the player receives these items to place them at the starting location.